code/src/steering_wheel_controller.cpp

```diff
@@ -0,0 +1,108 @@
+#include "../headers/steering_wheel_controller.hpp"
+
+r2d2::manual_control::steering_wheel_controller_c::steering_wheel_controller_c(
+    int controller_id, hwlib::pin_in &button1, hwlib::pin_in &button2,
+    hwlib::pin_in &button3, hwlib::pin_in &button4, hwlib::adc &steering_wheel,
+    hwlib::adc &pedals)
+    : controller_interface_c(controller_id),
+      button1(button1),
+      button2(button2),
+      button3(button3),
+      button4(button4),
+      steering_wheel(steering_wheel),
+      pedals(pedals) {
+}
+
+bool r2d2::manual_control::steering_wheel_controller_c::read() {
+    this->refresh();
+
+    return false;
+}
+
+unsigned char
+r2d2::manual_control::steering_wheel_controller_c::get_slider(sliders slider) {
+    int pedalsvalue = get_pedals();
+    switch (slider) {
+    case sliders::slider_l:
+        if (pedalsvalue > 0) {
+            return 0;
+        } else {
+            return pedalsvalue * -1;
+        }
+    case sliders::slider_r:
+        if (pedalsvalue < 0) {
+            return 0;
+        } else {
+            return pedalsvalue;
+        }
+    default:
+        return 0;
+    }
+}
+
+bool r2d2::manual_control::steering_wheel_controller_c::get_button(
+    buttons button) {
+    switch (button) {
```

Comment:
At first I though this was a switch in the function parameter, this means this is hard to read, please make this easier to read by either using the namespace as suggested above so that the parameter can be on the same line as the function or by tabbing the parameter at least once